src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+            updateManufacturerStatement.setString(2, manufacturer.getCountry());
+            updateManufacturerStatement.setLong(3, manufacturer.getId());
+            updateManufacturerStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create new manufacturer. Name = "
+                    + manufacturer.getName() + "; country = " + manufacturer.getCountry(), e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getManufacturer = "SELECT * FROM manufacturers_db WHERE id = ? "
+                + "AND is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacturerStatement = connection
+                         .prepareStatement(getManufacturer)) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            Manufacturer manufacturer = null;
+            if (resultSet.next()) {
+                manufacturer = parseManufacturer(resultSet, id);
+            }
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer by id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String selectQuery = "SELECT * FROM manufacturers_db WHERE is_deleted = false";
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllManufacturersStatement = connection
+                         .prepareStatement(selectQuery)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                Manufacturer manufacturer = parseManufacturer(resultSet, resultSet.getLong("id"));
+                allManufacturers.add(manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all data from database", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE manufacturers_db SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteManufacturerStatement = connection
+                         .prepareStatement(deleteQuery)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer by id = " + id, e);
+        }
+    }
+
+    private static Manufacturer parseManufacturer(ResultSet resultSet, Long id)
```

Comment:
and you can get id from `resultSet`
```suggestion
    private Manufacturer parseManufacturer(ResultSet resultSet, Long id)
```